### ISSUE TITLE
Bind & Call

### ISSUE THREAD

**Issue by robhuzzey** [philmeehan/promiseAdventure#2]
Hey,

I commented on Kev's issue already but I wanted to highlight a thing I noticed in the code that didn't make sense to me:
https://github.com/philmeehan/promiseAdventure/blob/master/separateObjectBindingThis.js#L24

Usually you would use `bind` on the invocation of the function rather than the definition... on that line `this` points to the module... is that expected behaviour?

> I must admit my ignorance of bind/call/apply

This helps me:
`bind` - returns a new function with the scope shifted to the scope passed in.
`call` - calls the function with the scope passed in
`apply` - same as call except you have to pass an array of arguments.

I would be very happy to sit with you & discuss all these things, maybe even just to bounce ideas off?

I hope that helps :)

**Comment by philmeehan** [comment]
Yep - I would have expected this to have been the object in question, as if I had called these functions without promises but as you saw in the chapter1 example it lost this.

By putting the bind here it keeps itself
